### PR TITLE
Files Wordlist Helper

### PR DESCRIPTION
**Pull request by xbenyx**
That is a simple helper when creating Wordlists. Just besides the upload button a helper that on expand, let you create a simple wordlist that on generate downloads a txt file. 

image

### CODE REVIEW

File: src/app/files/new-files/new-files.component.ts [hunk 1]
```diff
@@ -268,6 +271,16 @@ export class NewFilesComponent implements OnInit, OnDestroy {
     this.fileName = files[0].name;
   }
 
+  showHelp(): void {
+    const dialogRef = this.dialog.open(WordlisGeneratorComponent, {
+      width: '100%',
+      maxWidth: '100vw'
+    });
+    dialogRef.afterClosed().subscribe((result) => {
+      console.log('Dialog closed with result:', result);
```
Comment (itbf9) [inline src/app/files/new-files/new-files.component.ts:280]: This looks like debug output, remove this

File: src/app/shared/wordlist-generator/wordlist-generator.component.html [hunk 2]
```diff
@@ -0,0 +1,104 @@
+
+<grid-main>
+  <h2 mat-dialog-title class="h6">WordList Generator</h2>
+  <div class="content">
+    <form [formGroup]="form" (ngSubmit)="onSubmit()" >
+      <!-- Display added special dates -->
+      <div fxLayout="row" fxLayoutGap="10px">
+      <!-- Display added names -->
+      <div formArrayName="names">
```
Comment (itbf9) [inline src/app/shared/wordlist-generator/wordlist-generator.component.html:9]: Indentation is off

File: src/app/shared/wordlist-generator/wordlist-generator.component.html [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          <div fxLayout="column" fxLayoutAlign="start stretch" fxLayoutGap="10px">
+            <mat-form-field>
+              <mat-label>File Name</mat-label>
+              <input matInput formControlName="filename" name="filename" placeholder="Wordlist.txt" />
+              <mat-hint align="start">
+                {{ filenamehint }}
+              </mat-hint>
+            </mat-form-field>
+          </div>
+          <mat-checkbox formControlName="useSpecCharacters">Add special chars</mat-checkbox>
```
Comment (itbf9) [inline src/app/shared/wordlist-generator/wordlist-generator.component.html:88]: Maybe we should have text like these as variables instead? That way it is easy to implement support for other languages in the future.

**Review by itbf9 — APPROVED** [review]
Had a few minor comments, other than that it lgtm!

I do think the layout need some work, the dialog spans over the entire screen now, maybe we could have the checkboxes verically instead? Also the icons for uploading are a bit confusing, they suggest upload/download i think. And the icon for the helper is also a bit confusing, maybe there's a better one?

I approve so that you can merge when you have updated the code

File: src/app/shared/wordlist-generator/wordlist-generator.component.html (hunk 3 above)
Comment (xbenyx) [reply src/app/shared/wordlist-generator/wordlist-generator.component.html]: Changed to variables

**Pull request merged**